main_test.go

```diff
@@ -25,3 +31,148 @@ func TestCIRequirements(t *testing.T) {
 		}
 	}
 }
+
+type providerConfig struct {
+	providerType  string
+	username      string
+	password      string
+	url           string
+	defaultBranch string
+}
+
+var providers = []providerConfig{
+	{
+		providerType:  "azdo",
+		username:      "gitops-promotion",
+		password:      os.Getenv("AZDO_PAT"),
+		url:           os.Getenv("AZDO_URL"),
+		defaultBranch: "main",
+	},
+	{
+		providerType:  "github",
+		username:      "gitops-promotion",
+		password:      os.Getenv("GITHUB_TOKEN"),
+		url:           os.Getenv("GITHUB_URL"),
+		defaultBranch: "main",
+	},
+}
+
+func runCommand(command string, args ...string) (string, error) {
+	var cmdline = []string{"ze-binary", command}
```

Comment:
Helper functions usually start with `test` - `testRunCommand`. They also usually take `t *testing.T` and starts with `t.Helper()`.

Reply:
This is not a helper, tho. It actually wraps the execution of the code under test. I think that if we call `Helper()` here, we will "exclude" the entire test, I think?

Reply:
My understanding is that a function in a test file that is only used with tests is a test helper function - but that may be a personal preference more than anything and not something I see as a requirement. I usually name these functions `testFunction` with the `test`-prefix.

The reason `t.Helper()` exists is to not print information about that specific function when something fails and you know best if that's needed here. Linters usually remind on this.

You choose, feel free to set as resolved.

Reply:
Fixed.